src/Text/LLVM/AST.hs

```diff
@@ -347,6 +349,98 @@ isPointer PtrOpaque = True
 isPointer _         = False
 
 
+type TypeView = TypeView' Ident
+
+-- | Like `Type'`, but where the 'PtrTo' and 'PtrOpaque' constructors have been
+-- collapsed into a single 'PtrView' constructor. This provides a coarser notion
+-- of type equality than what `Type'` provides, which distinguishes the two
+-- types of pointers.
+--
+-- `TypeView'` is not used directly in any of the other AST types. Instead, it
+-- is primarly used as an auxiliary type to power the 'eqTypeModuloOpaquePtrs'
+-- and 'cmpTypeModuloOpaquePtrs' functions.
+data TypeView' ident
```

Comment:
It was pointed out to me that this patch exports the `TypeView'` data type by virtue of `Text.LLVM.AST` not having an explicit export list. This isn't what I intended, as `TypeView'` is meant to be internal to `eqTypeModuloOpaquePtrs` and `cmpTypeModuloOpaquePtrs` functions. We should add an explicit export list to `Text.LLVM.AST` to make sure that this actually happens.